apps/03TicTacToe.js

```diff
@@ -19,36 +19,70 @@ function printBoard() {
     console.log('1 ' + board[1].join(' | '));
     console.log('  ---------');
     console.log('2 ' + board[2].join(' | '));
-}
+};
 
 function horizontalWin() {
-    // Your code here
-}
+  return (board[0][0] === playerTurn && board[0][1] === playerTurn && board[0][2] === playerTurn) ||
+  (board [1][0] === playerTurn && board[1][1] === playerTurn && board[1][2] === playerTurn) ||
+  (board[2][0] === playerTurn && board[2][1] === playerTurn && board[2][2]=== playerTurn);
+
+};
 
 function verticalWin() {
-    // Your code here
-}
+  return (board[0][0] === playerTurn && board[1][0] === playerTurn && board[2][0] === playerTurn) ||
+  (board [0][1] === playerTurn && board[1][1] === playerTurn && board[2][1] === playerTurn) ||
+  (board[0][2] === playerTurn && board[1][2] === playerTurn && board[2][2]=== playerTurn);
+};
 
 function diagonalWin() {
-    // Your code here
-}
+  return (board[0][0] === playerTurn && board[1][1] === playerTurn && board[2][2] === playerTurn) ||
+  (board [0][2] === playerTurn && board[1][1] === playerTurn && board[2][0] === playerTurn);
+
+};
 
 function checkForWin() {
-    // Your code here
+  if (verticalWin() || horizontalWin() || diagonalWin()){
+  printBoard();
+  console.log ("Player" + " " + playerTurn + " " + "Wins!" );
+  cleanBoard();
 }
+  else {
+    return false;
+  }
+};
+
+
 
 function ticTacToe(row, column) {
-    // Your code here
+    if (board[row][column] === 'X' || board[row][column] === 'O') {
+      console.log('You must select an empty space! Please try again.');
+      return true;
+    }
+    else {
+      board[row][column] = playerTurn;
+      checkForWin();
+      playerTurn = (playerTurn === 'X') ? 'O' : 'X';
+    }
+  };
+// My 'should detect a win' test is failing even though the game works and clearly checks and delares a winner...
```

Comment:
Oh you saw this! nice. check my other comment around returning true in the checkForWin function!